plugin/formatting.py

```diff
@@ -59,13 +59,23 @@ class LspFormatDocumentCommand(LspTextCommand):
 
     capability = 'documentFormattingProvider'
 
+    def is_enabled(self, event: Optional[dict] = None) -> bool:
+        return super().is_enabled() or bool(self.session(LspFormatDocumentRangeCommand.capability))
+
     def run(self, edit: sublime.Edit, event: Optional[dict] = None) -> None:
         session = self.session(self.capability)
-        file_path = self.view.file_name()
-        if session and file_path:
-            session.send_request(
-                text_document_formatting(self.view),
-                lambda response: apply_response_to_view(response, self.view))
+        if session:
```

Comment:
Why no `file_name()` check anymore?

Reply:
That's already taken care of in `self.session(...)`.